src/NuGet.Core/NuGet.Build.Tasks.Pack/PackTaskLogic.cs

```diff
@@ -401,16 +401,11 @@ private void PopulateFrameworkReferences(PackageBuilder builder, LockFile assets
             foreach (var framework in assetsFile.PackageSpec.TargetFrameworks)
             {
                 var frameworkShortFolderName = framework.FrameworkName.GetShortFolderName();
+                tfmSpecificRefs.Add(frameworkShortFolderName, new HashSet<string>(ComparisonUtility.FrameworkReferenceNameComparer));
```

Comment:
Will this write out framework references for projects that don't have any?

It's not immediately obvious because that's probabyl handled in the packagemetadataxmlextensions or something like thaT?

Reply:
@nkolev92 Just to be clear, do we *want* this to write out empty frameworkreferences for projects that don't have any?

Reply:
I think it's ok if we don't do it. 
It doesn't provide much value. Framework references are like `lib` folders, unlike dependencies which influence the package graph. 

I'd verify what happens if there's a test with manually specified framework reference groups.

Reply:
Did you verify how the hand authored nuspec scenario works? 
How about whether anything get written if you have no framework references whatsoever?

Reply:
I verified it by hand. I'm just having trouble getting a good test going for it.

Reply:
I haven't done anything for the hand authored nuspec scenario yet. I did verify (by hand) that nothing gets written if there's no framework references and I'm working on getting tests for that going before moving to the nuspec.

Reply:
I think we should have tests for the nuspec scenario before + the no framework refs scenarios in this PR. 

The PR looks good overall, I just think we need more tests.

Reply:
I agree with Nikolche's comment above. It looks like `PackCommandRunner` decides whether to pack from `nuspec` or `csproj` file based on the `path` parameter. Considering scope of this change, I guess [`PackTaskLogicTests`](https://github.com/NuGet/NuGet.Client/blob/dev/test/NuGet.Core.Tests/NuGet.Build.Tasks.Pack.Test/PackTaskLogicTests.cs) would be the right class file to add tests.

https://github.com/NuGet/NuGet.Client/blob/a63b63cb76ef3aac7b60f6fe9e7fe404826afcfc/src/NuGet.Core/NuGet.Commands/CommandRunners/PackCommandRunner.cs#L96-L107

Reply:
Looks like this got merged without the added tests :)